File: internal/content/testdata/bench/java/merge-sorted-arrays.java
import java.util.*;

class Solution {
    public long[] merge_sorted_arrays(long[] arr1, long[] arr2) {
        long[] result = new long[arr1.length + arr2.length];
        int i = 0, j = 0, k = 0;
        while (i < arr1.length && j < arr2.length) {
            if (arr1[i] <= arr2[j]) result[k++] = arr1[i++];
            else result[k++] = arr2[j++];
        }
        while (i < arr1.length) result[k++] = arr1[i++];
        while (j < arr2.length) result[k++] = arr2[j++];
        return result;
    }
}

File: internal/content/testdata/bench/rust/merge-sorted-arrays.rs
fn merge_sorted_arrays(arr1: Vec<i64>, arr2: Vec<i64>) -> Vec<i64> {
    let mut result: Vec<i64> = Vec::new();
    let mut i = 0usize;
    let mut j = 0usize;
    while i < arr1.len() && j < arr2.len() {
        if arr1[i] <= arr2[j] {
            result.push(arr1[i]);
            i += 1;
        } else {
            result.push(arr2[j]);
            j += 1;
        }
    }
    while i < arr1.len() {
        result.push(arr1[i]);
        i += 1;
    }
    while j < arr2.len() {
        result.push(arr2[j]);
        j += 1;
    }
    result
}
